src/ingestion/ec2_describe_reserved_instance_modifications.py

```diff
@@ -0,0 +1,154 @@
+SELECT account:Id::string, account:Name
+FROM {AWS_ACCOUNTS_TABLE}
+WHERE timestamp = (
+  SELECT MAX(timestamp) FROM {AWS_ACCOUNTS_TABLE}
+);
+"""
+
+
+def get_accounts_list(sf_client):
+    res = sf_client.cursor().execute(GET_ACCOUNTS_QUERY).fetchall()
+    return res
+
+
+LOAD_INSTANCE_LIST_QUERY = f"""
+INSERT INTO {INSTANCES_TABLE} (timestamp, instance)
+SELECT '{{snapshotclock}}'::timestamp_ltz, PARSE_JSON(column1)
+FROM VALUES {{format_string}}
+"""
+
+
+def get_data_worker(account):
+    try:
+        ec2_session = get_aws_client(account)
+        instances = []
+        try:
+            ec2_regions = [region['RegionName'] for region in ec2_session.client('ec2').describe_regions()['Regions']]
+        except Exception as e:
+            log.error(f"ec2_describe_reserved_instance_modifications: account: {account[1]} exception: {e}")
+            return None
+        for region in ec2_regions:
+            try:
+                client = ec2_session.client('ec2', region_name=region)
+                paginator = client.get_paginator('describe_reserved_instances_modifications')
+                page_iterator = paginator.paginate()
+                region = [instance for page in page_iterator for instance in page['ReservedInstancesModifications']]
+                instances.extend(region)
+            except Exception as e:
+                log.error(f"ec2_describe_reserved_instance_modifications: account: {account[1]} exception: {e}")
+                return None
+        instance_list = [json.dumps({**instance,"AccountId":account[0]}, default=str) for instance in instances]
+        log.info(f"ec2_describe_reserved_instance_modifications: account: {account[1]} instances: {len(instance_list)}")
+        return instance_list
+    except Exception as e:
+        log.error(f"ec2_describe_reserved_instance_modifications: account: {account[1]} exception: {e}")
+        return None
+
+def get_data(accounts_list):
+    start = datetime.datetime.now()
+    instance_list_list = Pool(4).map(get_data_worker, accounts_list)
+    instance_list = [x for l in instance_list_list if l for x in l]
+    if instance_list:
+        sf_client = get_snowflake_client()
+        instance_groups = groups_of(15000,instance_list)
+        snapshot_time = datetime.datetime.utcnow().isoformat()
+        for group in instance_groups:
+            query = LOAD_INSTANCE_LIST_QUERY.format(
+                snapshotclock=snapshot_time,
+                format_string=", ".join(["(%s)"] * len(group)))
+            sf_client.cursor().execute(query, group)
+    end = datetime.datetime.now()
+    log.info(f"start: {start} end: {end} total: {(end - start).total_seconds()}")
+
+def main():
+    sf_client = get_snowflake_client()
+    current_time = datetime.datetime.now(datetime.timezone.utc)
+    last_time = sf_client.cursor().execute(f'SELECT max(timestamp) FROM {INSTANCES_TABLE}').fetchall()[0][0]
+    if last_time is None or (current_time - last_time).total_seconds() > 86400:
```

Comment:
this seems it would be easier to read if we put the `current_time` into the SQL -- it's probably not going to be a bug, but every person who reads this will stop and pause to think if it might be a bug